packages/cmcd-validator-library/README.md

```diff
@@ -22,57 +37,154 @@ You can install the package using npm:
 npm install @montevideo-tech/cmcd-validator
 ```
 
-### Usage
+## Usage
 
-Example with **CMCDQueryValidator**, **CMCDHeaderValidator**, **CMCDJsonValidator** functions:
+### How to import the library
+To import this library in your project, you can do it in the following way:
 
-```javascript
+```js
 import { CMCDQueryValidator, CMCDHeaderValidator, CMCDJsonValidator } from "@montevideo-tech/cmcd-validator";
+```
+
+### How to call the functions to validate CMCD
+The media client data can be sent by one of three means:
+* As a custom HTTP request header,
+* As a HTTP query argument,
+* As a JSON object independent of the HTTP object request.
+
+It is for this reason that the library has three main functions:
+
+* CMCDHeaderValidator
+* CMCDQueryValidator
+* CMCDJsonValidator
+
+### Input of the library
+Each of these functions receive the following parameters in the following order:
+
+* `query, header or json:` string where there are the CMCD parameters to check. 
+* `config:` json where you can define:
+  * `specific keys`: the list of CMCD keys that you want to check for any errors.
+    
+  * `custom keys`: personal customizable keys that you want to validate.
+    * `key`: name of the key
+    * `type`: type of the key (*bool, string, token, number*)
+    * `header`: in which CMCD header will your custom key be sent in case you use header.
```

Comment:
which CMCD header your custom key will be sent in, in case a header request is used.